charts/owncloud/templates/cronjob.yaml

```diff
@@ -0,0 +1,54 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: {{ include "owncloud.fullname" . }}-cronjob
+  labels:
+    {{- include "owncloud.labels" . | nindent 4 }}
+  annotations:
+    ignore-check.kube-linter.io/run-as-non-root : "ownCloud does not support non-root containers"
+    ignore-check.kube-linter.io/no-read-only-root-fs : "ownCloud need to write /etc/environment on startup"
+spec:
+  schedule: {{ .Values.owncloud.crondSchedule | quote }}
+  concurrencyPolicy: Forbid
+  jobTemplate:
+    spec:
+      parallelism: 1
+      backoffLimit: 0
+      template:
+        metadata:
+          annotations:
+            checksum/config: {{ print .Values.owncloud.configExtra | sha256sum }}
+        {{- with .Values.podAnnotations }}
+            {{- toYaml . | nindent 12 }}
+        {{- end }}
+          labels:
+            {{- include "owncloud.selectorLabels" . | nindent 12 }}
+        spec:
+          restartPolicy: Never
+          containers:
+          - name: {{ .Chart.Name }}-cronjob
+            securityContext:
+              {{- toYaml .Values.securityContext | nindent 14 }}
+            image: {{ template "owncloud.image" . }}
+            imagePullPolicy: {{ .Values.image.pullPolicy }}
+            envFrom:
+              - configMapRef:
+                  name: owncloud-env-config
+            command: ['/usr/bin/entrypoint', "/usr/bin/owncloud", "/usr/bin/cronjob"]
```

Comment:
Mixed single/double quotes.
How is this intended to work? Why does it need to run the entire startup scripts instead of just `/usr/bin/cronjob`?

Reply:
> Mixed single/double quotes.

Addressed in https://github.com/owncloud-docker/helm-charts/pull/45/commits/3759f8e3ea4b9c762f19eef22d96633371350e23. Worked nevertheless.

> How is this intended to work? Why does it need to run the entire startup scripts instead of just `/usr/bin/cronjob`?

This is the only way it works. This goes back to a discussion we had in July 2022. The `/usr/bin/entrypoint` script doesn't initialise ownCloud correctly thus you cannot run any `occ`-related commands as it will complain that ownCloud is not installed which is due to symlinks and env variables being missing. Back then it was decided, that fixing `/usr/bin/entrypoint` is too risky instead only a documentation should be added which resulted in https://github.com/owncloud/docs-server/pull/507.